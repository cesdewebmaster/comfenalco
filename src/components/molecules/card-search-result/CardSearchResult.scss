@import "../../../styles/variables.scss";
@import "../../../styles/mixins.scss";
@import "../../../styles/theme.scss";

.m-card-search-result {
  background-color: var(--color-card-bg);
  border-radius: 4px;
  box-shadow: 0px 3px 6px #0000001f;
  display: flex;
  flex-direction: column;
  width: 325px;
  height: 324px;
  overflow: hidden;

  .card-image {
    height: 130px;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  }

  .card-data {
    padding: 19px 18px 16px;
    flex-grow: 1;
    display: flex;
    flex-direction: column;

    .card-data__category {
      border-left: 2px solid;
      font-family: $roboto-bold;
      font-size: size(15);
      line-height: 20px;
      padding-left: 4px;
      margin-bottom: 7px;
    }

    .card-data__title {
      color: var(--color-blackred__white);
      font-family: $roboto-bold;
      font-size: size(15);
      line-height: 20px;
      margin-bottom: 5px;
    }

    .card-data__description {
      color: var(--color-blackred__white);
      font-size: size(13);
      line-height: 18px;
    }

    .card-data__cta {
      color: $magenta;
      display: flex;
      border: 2px solid $magenta;
      border-radius: 4px;
      font-family: $roboto-bold;
      height: 42px;
      justify-content: center;
      align-items: center;
      margin-top: auto;
      transition: all 0.2s ease;

      &:hover {
        background-color: $magenta;
        color: $white;
      }
    }
  }

  @include fromy($xx-large) {
    width: 308px;
    height: 349px;
  }
}
