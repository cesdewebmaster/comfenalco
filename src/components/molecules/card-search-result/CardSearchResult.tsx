import { Link } from 'gatsby';
import { SearchResult } from '../../../interfaces/Article.interface';
import { hasStartSlash } from '../../../utils/functions';
import './CardSearchResult.scss';

interface Props {
    resultData: SearchResult
}

const CardSearchResult = ({ resultData }: Props) => {

    return (
        <div className='m-card-search-result'>
            <div className='card-image' style={{ backgroundImage: `url(${resultData.miniatura})` }}></div>
            <div className='card-data'>
                <h3
                    className='card-data__category'
                    style={{ color: resultData.category.color, borderLeftColor: resultData.category.color }}
                >
                    {resultData.category.name}
                </h3>
                <h4 className='card-data__title'>{resultData.title}</h4>
                <p className='card-data__description'>{resultData.description}</p>
                <Link
                    className='card-data__cta'
                    to={hasStartSlash(resultData.slug)}
                    aria-label={`Enlace a ${resultData.title}`}
                >
                    Saber m√°s
                </Link>
            </div>
        </div>
    )
}

export default CardSearchResult