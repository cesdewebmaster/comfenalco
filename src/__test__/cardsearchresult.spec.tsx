import { render } from "@testing-library/react";
import CardSearchResult from "../components/molecules/card-search-result/CardSearchResult";

describe('Pruebas <CardSearchResult />', () => {

    const itemResult = {
        entryId: "xaseedfeb",
        title: "Afilíate como independiente",
        description: "Texto de introducción rápida al servicio ofrecido para personas y familias o empresas",
        slug: '/afiliaciones/afiliacion-independiente',
        category: {
            name: "Afiliaciones",
            color: "#007B72"
        },
        miniatura: "//images.ctfassets.net/0s9sb7ld7kac/5TRR4gy7orEGamtOqRAjFN/6ea6749c55e08e60b0b8f0ff5d174236/afiliaciones.png"
    }

    test('should be match with snapshot', () => {
        const { container } = render(
            <CardSearchResult
                resultData={itemResult} />
        );

        expect(container).toMatchSnapshot();
    });
})
