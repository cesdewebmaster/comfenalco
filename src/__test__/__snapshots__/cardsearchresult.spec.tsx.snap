// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <CardSearchResult /> should be match with snapshot 1`] = `
<div>
  <div
    class="m-card-search-result"
  >
    <div
      class="card-image"
      style="background-image: url(//images.ctfassets.net/0s9sb7ld7kac/5TRR4gy7orEGamtOqRAjFN/6ea6749c55e08e60b0b8f0ff5d174236/afiliaciones.png);"
    />
    <div
      class="card-data"
    >
      <h3
        class="card-data__category"
        style="color: rgb(0, 123, 114); border-left-color: #007B72;"
      >
        Afiliaciones
      </h3>
      <h4
        class="card-data__title"
      >
        Afilíate como independiente
      </h4>
      <p
        class="card-data__description"
      >
        Texto de introducción rápida al servicio ofrecido para personas y familias o empresas
      </p>
      <a
        aria-label="Enlace a Afilíate como independiente"
        class="card-data__cta"
        href="/afiliaciones/afiliacion-independiente"
      >
        Saber más
      </a>
    </div>
  </div>
</div>
`;
